src/main/java/com/frostwire/jlibtorrent/SessionManager.java

```diff
@@ -1119,6 +1119,10 @@ public void run() {
         t.start();
     }
 
+    public TorrentStats trackTorrentStats(TorrentHandle torrentHandle, long samplingIntervalInMs, long maxHistoryInMs) {
+        return new TorrentStats(this, torrentHandle, samplingIntervalInMs, maxHistoryInMs);
```

Comment:
however... (thinking...)
if we go the route of a single `AlertListener`, we need to have a way to ask for stats about a torrent.

Perhaps a good way to do this will cost us adding one more method in `SessionManager`, and it could be something like this:

```java
TorrentStats tStats = sessionManager.getTorrentStats(torrentHandle)
// if it's not being tracked, it should probably return `null`
// and then you use it like
int[] downloadSpeedSamples = tStats.get(TorrentStats.Metric.DownloadRate);
```

This way as a user, you don't need to manage anything, you just get your "one time use" sample array.